plonky2x/core/src/frontend/hash/blake2/curta.rs

```diff
@@ -116,7 +116,7 @@ impl<L: PlonkParameters<D>, const D: usize> Hash<L, D, 96, true, 4> for BLAKE2B
 
     fn hash(message: Vec<u8>) -> [Self::Integer; 4] {
         let mut num_message_chunks = (message.len() as u64 / 128) + 1;
-        if num_message_chunks % 128 == 0 {
+        if message.len() % 128 == 0 && !message.is_empty() {
             num_message_chunks -= 1;
         }
```

Comment:
nit: (message.len() as u64 + 127) / 128 is simpler

Fine leaving it this way as well, just add a comment.